File: py_orpheus/src/error_handler.py
import obj

class ErrorHandler(object):
    had_error = False

    @classmethod
    def error(cls, line: int, msg: str):
        cls.report(line, "", msg)

    @classmethod 
    def runtime_error(cls, msg: str):
        cls.error(-1, msg)
        return obj.Error(message = msg)

    @classmethod
    def report(cls, line: int, where: str, msg: str):
        if(line != -1): print("[" + str(line) + "] error " + where + ": " + msg)
        cls.had_error = True

File: orpheus/src/error_handler.cpp
#include "../include/error_handler.h"

bool ErrorHandler::had_error = false;

void ErrorHandler::Error(int line, const std::string& msg) {
    Report(line, msg);
}

void ErrorHandler::RuntimeError(const std::string& msg) {
    Error(-1, msg);
}

void ErrorHandler::Report(int line, const std::string& msg) {
    if(line != -1) {
        std::cout << "Error on line " << std::to_string(line) << ": " << msg << "\n";       
    }
    had_error = true;
}
